backend/src/main/kotlin/org/loculus/backend/controller/SubmissionController.kt

```diff
@@ -252,16 +256,30 @@ class SubmissionController(
     fun getReleasedData(
         @PathVariable @Valid organism: Organism,
         @RequestParam compression: CompressionFormat?,
+        @RequestHeader(value = HttpHeaders.IF_MODIFIED_SINCE, required = false) ifModifiedSince: Long?,
     ): ResponseEntity<StreamingResponseBody> {
         val headers = HttpHeaders()
         headers.contentType = MediaType.parseMediaType(MediaType.APPLICATION_NDJSON_VALUE)
         if (compression != null) {
             headers.add(HttpHeaders.CONTENT_ENCODING, compression.compressionName)
         }
 
-        val streamBody = streamTransactioned(compression) { releasedDataModel.getReleasedData(organism) }
+        var lastTime: Long? = null
 
-        return ResponseEntity(streamBody, headers, HttpStatus.OK)
+        transaction {
+            lastTime = UpdateTrackerTable
+                .selectAll() // Select all rows
+                .mapNotNull { it[UpdateTrackerTable.lastTimeUpdatedDbColumn] } // Extract non-null datetime values
+                .maxOrNull()?.toTimestamp() // Find the maximum value
+        }
```

Comment:
Actually we had the agreement that we don't put much logic in controller methods (especially database access). I would prefer if this was in a separate model/service class.